src/main/kotlin/org/neo4j/graphql/GraphQLExtensions.kt

```diff
@@ -72,17 +77,17 @@ fun GraphQLFieldsContainer.getValidTypeLabels(schema: GraphQLSchema): List<Strin
 }
 
 @Suppress("SimplifiableCallChain")
-fun GraphQLFieldsContainer.label(includeAll: Boolean = false) = when {
+fun GraphQLFieldsContainer.label(includeAll: Boolean = false, quote: Boolean = true) = when {
     this.isRelationType() ->
         (this as? GraphQLDirectiveContainer)
             ?.getDirective(DirectiveConstants.RELATION)
-            ?.getArgument(RELATION_NAME)?.value?.toJavaValue()?.toString()?.quote()
-                ?: this.name.quote()
+            ?.getArgument(RELATION_NAME)?.value?.toJavaValue()?.toString()?.let { if (quote) it.quote() else it  }
```

Comment:
when do we not have the quotes for a label?

btw. odd that in the tck-examples several non-critical labels/rel-types ones were quoted.

Reply:
isn't it easier to just have a separate (inner) function that returns the unquoted and then one that wraps it with .quote() this 3 times same condition makes the code less readable and maintainable.